pkg/object/mutation/annotation.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package mutation
+
+import (
+	"errors"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	Annotation = "config.kubernetes.io/apply-time-mutation"
+)
+
+func HasAnnotation(u *unstructured.Unstructured) bool {
```

Comment:
All these functions appear to duplicate `dependson` code.

Reply:
I refactored the dependson code to only return the value for the dependson annotation, rather than being generic and accepting the annotation key string. This way both ba be operated on similarly as if they had a common interface.

Reply:
"appear to duplicate" is because they're using the same conventional interface for consistency with different implementations.